### PR TITLE
Azure support

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: giantswarm/giantswarm#2652

Add support for azure credential

### CODE REVIEW

File: server/endpoint/creator/azure/azure.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package azure
+
+type Credential struct {
+	ClientID       string `json:"clientID"`
```
Comment (JosephSalisbury) [inline server/endpoint/creator/azure/azure.go]: IIRC, this needs to be like, `client_id`, and so on?

File: server/endpoint/creator/azure/azure.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package azure
+
+type Credential struct {
```
Comment (kopiczko) [inline server/endpoint/creator/azure/azure.go]: Should be AzureCredential to be on par

File: service/creator/service.go [hunk 3]
```diff
@@ -107,10 +108,21 @@ func (s *Service) Create(ctx context.Context, request Request) (Response, error)
...
 			aws.AwsoperatorArnKey: []byte(request.AWS.AwsOperatorARN),
-		},
+		}
+	case "azure":
+		secret.Data = map[string][]byte{
+			azure.ClientIDKey:       []byte(request.Azure.ClientID),
+			azure.ClientSecretKey:   []byte(request.Azure.SecretID),
+			azure.SubscriptionIDKey: []byte(request.Azure.SubscriptionID),
+			azure.TenantIDKey:       []byte(request.Azure.TenantID),
+		}
```
Comment (kopiczko) [inline service/creator/service.go]: How about `default:`?

Comment (TheoBrigitte) [reply service/creator/service.go]: added default case which fails with invalidProviderError

File: server/endpoint/creator/azure/azure.go (hunk 2 above)
Comment (TheoBrigitte) [reply server/endpoint/creator/azure/azure.go]: fixed

File: service/creator/service.go [hunk 4]
```diff
@@ -107,10 +108,23 @@ func (s *Service) Create(ctx context.Context, request Request) (Response, error)
...
+	default:
+		return Response{}, microerror.Maskf(invalidProviderError, "%q provider is not supported")
```
Comment (kopiczko) [inline service/creator/service.go]: You need to add `, request.Provider)`

**Pull request merged**